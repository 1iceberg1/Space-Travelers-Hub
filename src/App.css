* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
}

.navbar {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 20px 75px;
    border-bottom: 1.5px solid #bbbbbb;
}

.navbar .logo {
    display: flex;
    align-items: center;
    gap: 20px;
}

.navbar img {
    width: 55px;
}

.navbar h1 {
    font-size: 27px;
    font-weight: 500;
}

.navbar .navLink span a:-webkit-any-link {
    position: relative;
    text-decoration: none;
    font-size: 16px;
    color: #1d70ff;
    margin-right: 30px;
}

.navbar .navLink span:nth-child(2)::after {
    content: '';
  position: absolute;
  width: 1.2px;
  height: 20px;
  top: 5.2%;
  right: 12.5%;
  background-color: #555555;
}

.rocket-container {
    width: 90%;
    margin: 30px auto;
}

.rocket {
    display: flex;
    gap: 20px;
    margin: 25px 0;
}

.rocket img {
    width: 270px;
    object-fit: cover;
}

.rocket .rocket-detail h2 {
    font-weight: 500;
}

.rocket .rocket-detail p {
    margin: 8px 0;
}

.rocket .rocket-detail button {
    border: none;
    margin-top: 10px;
    font-size: 14px;
    border-radius: 5px;
    padding: 10px 15px;
    color: #fff;
    background-color: #3f6cff;
}

.rocket .rocket-detail .cancel-btn {
    color: #555555;
    background-color: #fff;
    border: 1px solid #555555;
}

.rocket .rocket-detail button:hover {
    cursor: pointer;
}

.reserve {
    font-size: 14px;
    font-weight: 500;
    color: #fff;
    background-color: #00bcbc;
    list-style: none;
    padding: 0px 4px 2px 4px;
    border-radius: 3px;
    margin-right: 10px;
}

/* My Profile */
.myprofile {
    width: 90%;
    margin: 30px auto;
    display: flex;
    justify-content: space-between;
}

.myprofile h2 {
    font-size: 25px;
    font-weight: 400;
}

.my_missions {
    width: 49%;
}

.myrockets {
    width: 49%;
}

.myrockets ul {
    width: 70%;
    margin-top: 20px;
    border: 1px solid #bbbbbb;
    border-radius: 5px;
}
* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.navbar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 20px 75px;
  border-bottom: 1.5px solid #bbb;
}

.navbar .logo {
  display: flex;
  align-items: center;
  gap: 20px;
}

.navbar img {
  width: 55px;
}

.navbar h1 {
  font-size: 27px;
  font-weight: 500;
}

.navbar .navLink span a:-webkit-any-link {
  position: relative;
  text-decoration: none;
  font-size: 16px;
  color: #1d70ff;
  margin-right: 30px;
}

.navbar .navLink span:nth-child(2)::after {
  content: '';
  position: absolute;
  width: 1.2px;
  height: 20px;
  top: 5.2%;
  right: 12.5%;
  background-color: #555;
}

.rocket-container {
  width: 90%;
  margin: 30px auto;
}

.rocket {
  display: flex;
  gap: 20px;
  margin: 25px 0;
}

.rocket img {
  width: 270px;
  object-fit: cover;
}

.myprofile h2 {
  font-size: 25px;
  font-weight: 400;
}

.rocket .rocket-detail h2 {
  font-weight: 500;
}

.rocket .rocket-detail p {
  margin: 8px 0;
}

.rocket .rocket-detail button {
  border: none;
  margin-top: 10px;
  font-size: 14px;
  border-radius: 5px;
  padding: 10px 15px;
  color: #fff;
  background-color: #3f6cff;
}

.rocket .rocket-detail .cancel-btn {
  color: #555;
  background-color: #fff;
  border: 1px solid #555;
}

.rocket .rocket-detail button:hover {
  cursor: pointer;
}

.reserve {
  font-size: 14px;
  font-weight: 500;
  color: #fff;
  background-color: #00bcbc;
  list-style: none;
  padding: 0 4px 2px 4px;
  border-radius: 3px;
  margin-right: 10px;
}

/* My Profile */
.myprofile {
  width: 90%;
  margin: 30px auto;
  display: flex;
  justify-content: space-between;
}

.my_missions {
  width: 49%;
}

.myrockets {
  width: 49%;
}

.myrockets ul {
  width: 70%;
  margin-top: 20px;
  border: 1px solid #bbb;
  border-radius: 5px;
}

.myrockets ul li {
  list-style: none;
  padding: 20px;
  border-bottom: 1px solid #bbb;
}
