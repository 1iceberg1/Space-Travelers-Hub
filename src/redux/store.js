import { configureStore } from '@reduxjs/toolkit';
import rocketsSlice from './rockets/rocketsSlice';
import missionsSlice from './missions/missionsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsSlice,
    missions: missionsSlice,
  },
});

export default store;
