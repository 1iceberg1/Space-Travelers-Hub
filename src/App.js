import {
  BrowserRouter, Routes, Route,
} from 'react-router-dom';
import './App.css';
import Layout from './layout/Layout';
import Rockets from './layout/Rockets';
import MyProfile from './layout/MyProfile';
import Missions from './layout/Missions';

const App = () => (
  <BrowserRouter>
    <Routes>
      <Route path="/" element={<Layout />}>
        <Route index element={<Rockets />} />
      </Route>
    </Routes>
  </BrowserRouter>
);

export default App;
