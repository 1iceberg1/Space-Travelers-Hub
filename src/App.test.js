import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from './redux/store';
import App from './App';

test('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <App />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
