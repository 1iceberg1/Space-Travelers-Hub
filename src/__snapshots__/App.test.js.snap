// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <nav
    className="navbar"
  >
    <nav
      className="logo"
    >
      <img
        alt="logo"
        src="planet.png"
      />
      <h1>
        Space Travelers' Hub
      </h1>
    </nav>
    <nav
      className="navLink"
    >
      <span>
        <a
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </span>
      <span>
        <a
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </span>
      <span>
        <a
          href="/myprofile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </span>
    </nav>
  </nav>,
  <div
    className="rocket-container"
  />,
]
`;
